### DISCUSSION TITLE
Online routing  - openrouteservice not working?

### DISCUSSION THREAD

**Discussion by ERYpTION** [osmandapp/OsmAnd#12568] [General]
I have unsuccessfully tried setting up online routing for openrouteservice. I have entered an API key in the app navigation settings and according to the route calculation test in the settings, it works OK. However when I try to do an actual route calculation all I get is the pop-up message ‘Route is empty’. I have tried with several different addresses.

Has anyone successfully used openrouteservice for online routing in OsmAnd?

I posted the same question at Openrouteservice forum and got a reply from one of the developers. But I am not a coder myself so do not know if it is a solution: the https://ask.openrouteservice.org/t/osmand-route-is-empty/3088

**Comment by joker234** [comment]
Hi @ERYpTION, as mentioned in the ask forum, I created #12573 that should solve the basic problem and allow to use the openrouteservice within OsmAnd.
